### ISSUE TITLE
Multiple domains on multisite

### ISSUE THREAD

**Issue by JoryHogeveen** [polylang/polylang#342]
## Prerequisites

- [x] This is not a usage question (Those should be directed to the [community supported forum](https://wordpress.org/support/plugin/polylang), unless this is a question about Polylang Pro in which case you should [use the helpdesk](https://polylang.pro/support/)).
- [x] I have searched for similar issues in both open and closed tickets and cannot find a duplicate.
- [x] The issue still exists against the latest `master` branch of Polylang and the latest WordPress version.
- [x] This bug happens with only Polylang plugin active
- [x] This bug happens with a default WordPress theme active
- [x] I can reproduce this bug consistently

## Steps to reproduce the issue

1. Convert installation to a multisite
2. Configure different domains for languages
3. Site always redirects to main site domain (as configured in the MU settings)

## Expected behavior and actual behavior

When configuring Polylang with different domains for each language on a multisite it doesn't work. A while ago you could use the Domain Mapping plugin but since this plugin isn't supported for a long time (any won't be supported at all anymore) I'm not keen on using that from now on. It might be nice if Polylang could prevent the redirect itself.

**Comment by Chouby** [comment]
Which plugin are you using for the multisite domain mapping? Polylang should be compatible with https://wordpress.org/plugins/wordpress-mu-domain-mapping/.

**Comment by JoryHogeveen** [comment]
Hi @Chouby See my description:

> A while ago you could use the Domain Mapping plugin but since this plugin isn't supported for a long time (any won't be supported at all anymore) I'm not keen on using that from now on.

**Comment by Chouby** [comment]
Do you mean that you are not using any domain mapping plugin? Is there an official doc for this?

**Comment by JoryHogeveen** [comment]
Hi @Chouby,

> Do you mean that you are not using any domain mapping plugin?

Yes, I only know of the MU Domain Mapping plugin and since that one isn't supported for more than 3 years I can't find any other options for multisite/network installations.

> Is there an official doc for this?

WordPress doesn't support linking multiple domains to the same blog ID as far as I know.

**Comment by JoryHogeveen** [comment]
Might be an idea to embed such functionality inside Polylang?
Let me know if you would be interested in that, I can take a look at MU Domain Mapper and find a clean way to port it to Polylang when a multi-domain setup is chosen using site metadata or something.

**Comment by Chouby** [comment]
A quick search directed me to https://wordpress.org/support/article/wordpress-multisite-domain-mapping/. I missed that change introduced in WP 4.5.

**Comment by JoryHogeveen** [comment]
Yep, but this unfortunately doesn't support multiple domains for a single blog/site.
Previously (with the old domain mapping plugin) you could create an "alias" without redirecting and thus the website would work on multiple domains so Polylang could take it from there. Such a feature isn't included in 4.5+ but the plugin still got abandoned.

**Comment by Chouby** [comment]
For me the plugin https://wordpress.org/plugins/wordpress-mu-domain-mapping/ is still working correctly, although it's no been updated for a long time.

I had a look at how we could add more domains per blog (using the naming of WP here). It looks like this could be *maybe* done using the filter `pre_get_site_by_path`. I did not investigated more as the issue is that this filter is not accessible, not even to a mu-plugin. So we come back to the file `sunrise.php`, used by the MU domain mapping plugin.

See also https://core.trac.wordpress.org/ticket/39764

**Comment by JoryHogeveen** [comment]
Hmm I get the problem.
MU Domain Mapping is still working but I only wonder for how long.. I'll think about this some more. If you have any ideas I'm all ears!

**Comment by cyrillbolliger** [comment]
Has there been any progress concerning this issue?

**Comment by JoryHogeveen** [comment]
Hi @cyrillbolliger 
Nope, still using the MU Domain Mapping plugin.

**Comment by cyrillbolliger** [comment]
Thanks for your rapid response @JoryHogeveen. As using an unmaintained plugin was no option for us, I've worked my way around it using `sunrise.php`. I created a small gist, in case it helps you or someone else:

:arrow_right: [WordPress Multisite: Map multiple domains to single blog / subsite](https://gist.github.com/cyrillbolliger/0fc8f8d72273e5227890dfcf9d6c017a#wordpress-multisite-map-multiple-domains-to-single-blog--subsite)
